File: widgetssdk/src/main/java/com/glia/widgets/core/engagement/domain/GliaEndEngagementUseCase.java
package com.glia.widgets.core.engagement.domain;

import com.glia.widgets.core.engagement.GliaEngagementRepository;

public class GliaEndEngagementUseCase {

    private final GliaEngagementRepository engagementRepository;

    public GliaEndEngagementUseCase(GliaEngagementRepository engagementRepository) {
        this.engagementRepository = engagementRepository;
    }

    public void execute() {
        engagementRepository.endEngagement();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/engagement/domain/GliaEndEngagementUseCase.kt
package com.glia.widgets.core.engagement.domain

import com.glia.widgets.chat.data.ChatScreenRepository
import com.glia.widgets.core.engagement.GliaEngagementRepository

class GliaEndEngagementUseCase(
    private val engagementRepository: GliaEngagementRepository,
    private val chatScreenRepository: ChatScreenRepository
) {
    operator fun invoke() {
        engagementRepository.endEngagement()
        chatScreenRepository.isFromCallScreen = false
    }
}
